evalml/pipelines/components/transformers/imputers/time_series_imputer.py

```diff
@@ -0,0 +1,199 @@
+        """
+        forwards_fill_cats = (
+            ["category", "boolean"]
+            if self.parameters["categorical_impute_strategy"] == "forwards_fill"
+            else []
+        )
+        backwards_fill_cats = (
+            ["category", "boolean"]
+            if self.parameters["categorical_impute_strategy"] == "backwards_fill"
+            else []
+        )
+        interpolation_cats = []
+        if self.parameters["numeric_impute_strategy"] == "forwards_fill":
+            forwards_fill_cats.append("numeric")
+        elif self.parameters["numeric_impute_strategy"] == "backwards_fill":
+            backwards_fill_cats.append("numeric")
+        else:
+            interpolation_cats.append("numeric")
+
+        X = infer_feature_types(X)
+        forwards_cols = list(
+            X.ww.select(forwards_fill_cats, return_schema=True).columns
+        )
+        backwards_cols = list(
+            X.ww.select(backwards_fill_cats, return_schema=True).columns
+        )
+        interpolation_cols = list(
+            X.ww.select(interpolation_cats, return_schema=True).columns
+        )
+
+        nan_ratio = X.ww.describe().loc["nan_count"] / X.shape[0]
+        self._all_null_cols = nan_ratio[nan_ratio == 1].index.tolist()
+
+        X_forwards = X[[col for col in forwards_cols if col not in self._all_null_cols]]
+        if len(X_forwards.columns) > 0:
+            self._forwards_cols = X_forwards.columns
+
+        X_backwards = X[
+            [col for col in backwards_cols if col not in self._all_null_cols]
+        ]
+        if len(X_backwards.columns) > 0:
+            self._backwards_cols = X_backwards.columns
+
+        X_interpolate = X[
+            [col for col in interpolation_cols if col not in self._all_null_cols]
+        ]
+        if len(X_interpolate.columns) > 0:
+            self._interpolate_cols = X_interpolate.columns
+
+        if y is not None:
+            y = infer_feature_types(y)
+            if y.isnull().any():
+                self._impute_target = self.parameters["target_impute_strategy"]
+
+        return self
+
+    def transform(self, X, y=None):
+        """Transforms data X by imputing missing values using specified timeseries-specific strategies. 'None' values are converted to np.nan before imputation and are treated as the same.
+
+        Args:
+            X (pd.DataFrame): Data to transform
+            y (pd.Series, optional): Ignored.
+
+        Returns:
+            pd.DataFrame: Transformed X
+        """
+        X = infer_feature_types(X)
+        if len(self._all_null_cols) == X.shape[1]:
+            df = pd.DataFrame(index=X.index)
+            df.ww.init()
+            return df, y
+
+        X_no_all_null = X.ww.drop(self._all_null_cols)
+
+        if self._forwards_cols is not None:
+            X_forward = X.ww[self._forwards_cols.tolist()]
+            imputed = X_forward.pad()
+            imputed = imputed.bfill()  # Fill in the first value, if missing
```

Comment:
Why isn't this `imputed.ffill()` ?

Reply:
Ah, I think I misunderstood what `pad` was doing!

Reply:
Rather than calling `bfill` or `pad` on the entire dataset just to fill in the first/last value, I wonder if we could do something more targeted to prevent having to create a whole new copy of the data.

Reply:
I played around with some implementation options to just target the first/last row, but everything I tried took significantly longer than running `bfill` or `pad` on the whole dataset. If we want to avoid creating a copy of the data, we could use `inplace=True` instead.

Reply:
Awesome, let's use `inplace`!